tests/test_core.py

```
@@ -4580,6 +4655,42 @@ async def test_ci_fails_all_attempts_exhausted(self) -> None:
         assert "Tests" in result.summary
         assert mock_invoke.await_count == 2
 
+    async def test_low_budget_fraction_stops_retry_without_failing(self) -> None:
+        """Below the 20% threshold, CI fix retries stop but the step reports success."""
+        from sova.config.models import AgentConfig
+        from sova.core.steps.monitor_ci import MonitorCIStep
+
+        config = ProjectConfig(agent=AgentConfig(max_budget=Decimal("10.00")))
+        ctx = _make_ctx(pr_number=10, branch_name="feat/test", worktree_dir=Path("/tmp/wt"), config=config)
```

Ruff (0.16.3)

[error] Probable insecure usage of temporary file or directory: "/tmp/wt"

(S108)